src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,215 @@
+                cars.add(getCar(resultSet));
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Can't get all cars from cars table.", throwable);
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateQuery = "UPDATE cars SET model = ?, manufacturer = ?"
+                + " WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateStatement = connection.prepareStatement(updateQuery)) {
+            updateStatement.setString(1, car.getModel());
+            updateStatement.setObject(2, car.getManufacturer().getId());
+            updateStatement.setObject(3, car.getId());
+            if (updateStatement.executeUpdate() > 0) {
+                deleteDrivers(car);
+                insertDrivers(car);
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Can't update car " + car
+                + "in the cars table.", throwable);
+        }
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteQuery = "UPDATE cars SET is_deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteStatement = connection.prepareStatement(deleteQuery)) {
+            deleteStatement.setObject(1, id);
+            return deleteStatement.executeUpdate() > 0;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Can't delete car with id "
+                    + id + ".", throwable);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getAllByDriverQuery = "SELECT c.id, c.model, c.manufacturer, "
+                + "m.id, m.name, m.country "
+                + "FROM cars c JOIN manufacturers m ON c.manufacturer = m.id "
+                + "JOIN cars_drivers cd ON c.id = cd.car_id "
+                + "WHERE cd.driver_id = ? AND c.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement getAllByDriverStatement =
+                            connection.prepareStatement(getAllByDriverQuery)) {
+            getAllByDriverStatement.setLong(1, driverId);
+            ResultSet resultSet = getAllByDriverStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Can't get all cars by driver with id "
+            + driverId + ".",throwable);
+        }
+        if (!cars.isEmpty()) {
+            cars.forEach(car -> car.setDrivers(getDriversListForCar(car.getId())));
+        }
+        return cars;
```

Comment:
```suggestion
        }
            cars.forEach(car -> car.setDrivers(getDriversListForCar(car.getId())));
        return cars;
```
forEach doesn't need check for isEmpty()